torchvision/ops/ciou_loss.py

```diff
@@ -0,0 +1,84 @@
+import math
+
+import torch
+
+
+def ciou_loss(
+    boxes1: torch.Tensor,
+    boxes2: torch.Tensor,
+    reduction: str = "none",
+    eps: float = 1e-7,
+) -> torch.Tensor:
+
+    """
+    Original Implementation from
+    https://github.com/facebookresearch/detectron2/blob/main/detectron2/layers/losses.py
+
+
+    Args:
+        boxes1 : (Tensor[N, 4] or Tensor[4]) first set of boxes
+        boxes2 : (Tensor[N, 4] or Tensor[4]) second set of boxes
+        reduction : (string, optional) Specifies the reduction to apply to the output:
+            ``'none'`` | ``'mean'`` | ``'sum'``. ``'none'``: No reduction will be
+            applied to the output. ``'mean'``: The output will be averaged.
+            ``'sum'``: The output will be summed. Default: ``'none'``
+        eps : (float, optional): small number to prevent division by zero. Default: 1e-7
+
+    Reference:
+
+    Complete Intersection over Union Loss (Zhaohui Zheng et. al)
+    https://arxiv.org/abs/1911.08287
+
+    """
+
+    x1, y1, x2, y2 = boxes1.unbind(dim=-1)
+    x1g, y1g, x2g, y2g = boxes2.unbind(dim=-1)
+
+    if (x2 < x1).all():
+        raise ValueError("x1 is larger than x2")
+    if (y2 < y1).all():
+        raise ValueError("y1 is larger than y2")
```

Comment:
We shouldn't put an assertion here to avoid slowdowns due to GPU<=>CPU syncs.

Reply:
Okay I will do that.

Reply:
A doubt - 

So the user has to manually check if there are no bad inputs to this function, right?  should we have some warning message in that scenario?

Reply:
Understood thanks!

Reply:
The problem is you can't have a warning unless you check and if you check you slowdown the execution significantly. The user needs to make sure that they don't provide bad inputs. If they want the warnings the recommendation is to wrap this method on their end and perform those checks.

You can see similar discussions at #5322, #5691 and #5684.

Reply:
Damn, Github **really** needs to implement a feature to info you that others are writing comments at the same time. It happens all the time that we are posting concurrently the same things. lol :D